doc/man3/hostlist_create.rst

```diff
@@ -0,0 +1,150 @@
+
+  int hostlist_delete (struct hostlist *hl, const char *hosts);
+
+  int hostlist_count (struct hostlist *hl);
+
+  void hostlist_sort (struct hostlist *hl);
+
+  void hostlist_uniq (struct hostlist *hl);
+
+  const char *hostlist_first (struct hostlist *hl);
+
+  const char *hostlist_last (struct hostlist *hl);
+
+  const char *hostlist_next (struct hostlist *hl);
+
+  const char *hostlist_current (struct hostlist *hl);
+
+  int hostlist_remove_current (struct hostlist *hl);
+
+Link with :command:`-lflux-hostlist`.
+
+DESCRIPTION
+===========
+
+A hostlist is an ordered list of hostnames that can be encoded as a
+compact string when the hostnames contain numerical suffixes.
+
+:func:`hostlist_create` creates an empty hostlist.
+
+:func:`hostlist_destroy` destroys a hostlist.
+
+:func:`hostlist_decode` converts an RFC 29 hostlist string into a hostlist.
+The caller must free the result with :func:`hostlist_destroy`.
+
+:func:`hostlist_encode` converts a hostlist into an RFC 29 hostlist string.
+The caller must free the result.
+
+:func:`hostlist_copy` makes a copy of a hostlist.
+
+:func:`hostlist_append` decodes an RFC 29 hostlist string and appends its
+hostnames to another hostlist.
+
+:func:`hostlist_append_list` appends the hostnames of a hostlist to another
+hostlist.
+
+:func:`hostlist_nth` returns the hostname at a zero-origin index and updates
+the cursor.
+
+:func:`hostlist_find` finds the zero-origin index of a hostname and updates
+the cursor.
+
+:func:`hostlist_delete` deletes an RFC 29 hostlist string from a hostlist.
+
+:func:`hostlist_count` returns the number of hostnames in a hostlist.
+
+:func:`hostlist_sort` sorts a hostlist object.
+
+:func:`hostlist_uniq` sorts a hostlist object, then removes duplicate
+hostnames.
+
+:func:`hostlist_first` returns the first hostname and updates the cursor.
+
+:func:`hostlist_last` returns the last hostname and updates the cursor.
+
+:func:`hostlist_next` returns the next hostname relative to the cursor and
+updates the cursor.
+
+:func:`hostlist_current` returns current hostname relative to the cursor.
+
+:func:`hostlist_remove_current` removes the current hostname.
+
+RETURN VALUE
+============
+
+:func:`hostlist_create`, :func:`hostlist_decode`, and :func:`hostlist_copy`
+return a hostlist on success which must be freed with :func:`hostlist_destroy`.
+On failure, NULL is returned with :var:`errno` set.
+
+:func:`hostlist_encode` returns a string on success that must be freed.
+On failure, NULL is returned with :var:`errno` set.
+
+:func:`hostlist_append`, :func:`hostlist_append_list`, :func:`hostlist_delete`,
+and :func:`hostlist_remove_current` return a count on success.
+On failure, -1 is returned with :var:`errno` set.
+
+:func:`hostlist_count` returns a count.  If the hostlist is invalid, zero
+is returned.
+
+:func:`hostlist_find` returns an index on success.
+On failure, -1 is returned with :var:`errno` set.
+
+:func:`hostlist_sort` and :func:`hostlist_uniq` return nothing.
```

Comment:
For `hostlist_first`, `hostlist_next`, `hostlist_last`, and `hostlist_nth`, it should be noted that the return value is only valid until the next call which modifies the cursor (all the above plus `hostlist_find`)

Reply:
Just pushed fixes.  I made another editing pass on hostlist_create(3) so please give it another peek when you have a chance.
- expanded the intro paragraph a bit
- added a paragraph about the cursor and validity of returned hostnames at the top
- reworded some function descriptions so that it's more clear what's happening with the cursor
- noted that `hostlist_sort()`, `hostlist_uniq()`, `hostlist_delete()` also may update the cursor (I think?  Please confirm)

Reply:
Looks good to me!